File: internal/indexer/testdata/python/views/critical_user_model_view.py
from models.critical_user_model import CriticalUserModel, create_critical_user_model


class CriticalUserModelRenderer:
    """Renders a CriticalUserModel for display."""

    def __init__(self, model: CriticalUserModel):
        self.model = model

    def render(self) -> str:
        return self.model.display()


def render_critical_user_model(m: CriticalUserModel) -> str:
    """Formats a CriticalUserModel as a display string."""
    return f"{m.name} <{m.email}>"


def build_critical_user_model(name: str, email: str) -> CriticalUserModel:
    """Creates a CriticalUserModel from the given parts."""
    return create_critical_user_model(name, email)

File: internal/indexer/testdata/ts/views/criticalUserModelView.ts
import { CriticalUserModel, createCriticalUserModel } from "../models/criticalUserModel";

/** CriticalUserModelRenderer renders a CriticalUserModel for display. */
export class CriticalUserModelRenderer {
  model: CriticalUserModel;

  constructor(model: CriticalUserModel) {
    this.model = model;
  }

  render(): string {
    return this.model.display();
  }
}

/** renderCriticalUserModel formats a CriticalUserModel as a display string. */
export function renderCriticalUserModel(m: CriticalUserModel): string {
  return m.name + " <" + m.email + ">";
}

/** buildCriticalUserModel creates a CriticalUserModel from parts. */
export function buildCriticalUserModel(name: string, email: string): CriticalUserModel {
  return createCriticalUserModel({ name, email });
}
